DABI.js

```diff
@@ -0,0 +1,321 @@
+    } else if (doc.title.trim().indexOf("DABI: Rechercheergebnis") == 0) {
+        var keinTreffer = doc.getElementsByTagName("br")[7].nextSibling.data.indexOf("Keine Treffer für die Suche nach:")
+        if (keinTreffer == -1) {
+            return "multiple";
+        }
+    }
+}
+
+function doWeb(doc, url) {
+    var ids = [];
+
+    //If Statement checks if page is a Search Result, then saves requested Items
+    if (detectWeb(doc, url) == "multiple") {
+        Z.selectItems(getSearchResults(doc), function(items) {
+            if (!items) return true;
+            for (var i in items) {
+                ids.push(i);
+            }
+        });
+    } else if (detectWeb(doc, url) == "journalArticle") {
+        //saves single page items
+        ids = [url];
+    }
+    ZU.processDocuments(ids, scrape);
+}
+
+
+function getSearchResults(doc) {
+    var trs = doc.getElementsByTagName("tr"),
+        tds = null,
+        items = {};
+
+    for (var i = 1; i < trs.length; i++) {
+        tds = trs[i].getElementsByTagName("td");
+        for (var n = 0; n < tds.length; n++) {
+            var url = doc.location.origin + "/cgi-bin/dabi/" + tds[0].firstChild.getAttribute("href"),
+                author = tds[1].innerHTML,
+                title = tds[2].innerHTML.replace(/<br>/g, '. ');
+
+            if (!author == '') {
+                var item = author.replace(/; <br>.*/, ' et al.') + ": " + title;
+            } else {
+                var item = title;
+            }
+            if (!item || !url) continue;
+
+            items[url] = item;
+        }
+    }
+    return items;
+}
+
+function scrape(doc, url) {
+    var newItem = new Zotero.Item('journalArticle');
+    var trs = doc.getElementsByTagName("tr"),
+        headers,
+        contents,
+        items = {};
+
+    //For Loop to populate "items" Object and save tags to an Array.
+    for (var i = 0; i < trs.length; i++) {
+        headers = trs[i].getElementsByTagName("th")[0].textContent;
+        contents = trs[i].getElementsByTagName("td")[0].innerHTML;
+
+        items[headers.replace(/\s+/g, '')] = contents.trim();
+    }
+
+    //set url to fulltext resource, if present; else to database item
+    if (items["URL"] == '') {
+        newItem.url = url;
+    } else {
+        var link = doc.createElement('a');
+        link.innerHTML = items["URL"];
+        newItem.url = link.firstChild.getAttribute("href");
+
+        if (/\.pdf(#.*)?$/.test(newItem.url)) {
+            newItem.attachments = [{
+                url: newItem.url,
+                title: "DABI Full Text PDF",
+                mimeType: "application/pdf"
+            }];
+        }
+    }
+
+
+    //Formatting and saving "title" fields
+    if (items["Titel"]) {
+        newItem.title = items["Titel"].replace(/\*/g, '');
+        var short = newItem.title.replace(/^\W?(?:Die |Der |Das |\.{3}\s?)?/, '');
+        short = short.replace(/\W?(,|:|\?|!|\.|\"|\')/, '').split(' ').slice(0, 6).join(' ');
```

Comment:
instead of a capturing group (`(,|:|\?|!|\.|\"|\')`), use character group `[,:?!."']`

Reply:
changed